### PR TITLE
Pass vllm args as extra arguments

### PR DESCRIPTION
**Pull request by tiran**
Additional vLLM args are now passed as click extra args. They are more convenient to use than passing each extra argument as a separate arg.

```console
ilab model serve ... --backend=vllm -- --dtype=auto --enable-lora
```

Also clean up unnecessary parameter handling in llama-cpp path.

**Checklist:**

- [x] **Commit Message Formatting**: Commit titles and messages follow guidelines in the
  [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/#summary).
- [ ] [Changelog](https://github.com/instructlab/instructlab/blob/main/CHANGELOG.md) updated with breaking and/or notable changes for the next minor release.
- [x] Documentation has been updated, if necessary.
- [x] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Comment by cdoern** [comment]
This is going to clobber https://github.com/instructlab/instructlab/pull/1634 which takes a different approach config wide. Basically I am adding `additional_args` to serve, train, etc. 

The current `vllm_args` will become a dict (probably?) of flag name -> value pairs. Some vllm args will earn a spot in the actual vllm config yaml if we deem them important enough and commonly edited.

I think maybe these two PRs can coexist? But just wanted to voice that the `additional_args` is kind of needed for stuff like training and I am trying to make a unified config language.

**Comment by cdoern** [comment]
The issue here is:

I believe commonly edited vllm args should live in the cfg and therefore inherit defaults in the same way all other flags do.

`additional_args` is as far as I think we should go with "implicit" option passing. 

Maybe we can keep this worklfow for the `vllm_additional_args` implicit vllm options as outlined out in #1634 but we should also have some as their own flags IMO

File: src/instructlab/model/serve.py [hunk 1]
```diff
@@ -129,6 +121,14 @@ def serve(
         )
     elif backend == backends.VLLM:
         # Instantiate the vllm server
+        if ctx.args:
+            # extra click arguments after "--"
+            vllm_args = ctx.args
+        elif ctx.obj.config.serve.vllm.vllm_args:
```
Comment (cdoern) [inline src/instructlab/model/serve.py:127]: if the options are still in the cfg, then I don't think this impacts #1634. I honestly like this new way of dealing w/ the flags.

File: src/instructlab/model/serve.py [hunk 2]
```diff
@@ -66,16 +67,6 @@
         "Automatically detected based on the model file properties.\n"
     ),
 )
-@click.option(
```
Comment (cdoern) [inline src/instructlab/model/serve.py:69]: we should probably document somewhere if we support ALL vllm args

**Review by cdoern — APPROVED** [review]
took another look, I don't think this is a huge blocker for #1634 so this gets my +1

**Comment by russellb** [comment]
a question -- previously you could do something like this in `config.yaml`

```yaml
serve:
  ...
  backend: "vllm"
  backend_args:
    vllm_args:
    - "custom_arg"
```

I don't care about the syntax, just the ability to put the args in the config file. Is that still supported in some way?

**Comment by leseb** [comment]
> a question -- previously you could do something like this in `config.yaml`
> 
> ```yaml
> serve:
>   ...
>   backend: "vllm"
>   backend_args:
>     vllm_args:
>     - "custom_arg"
> ```
> 
> I don't care about the syntax, just the ability to put the args in the config file. Is that still supported in some way?

`backend_args` does not exist, but I see what you mean. All of that I left unchanged, only the CLI flags are touched. So if you had that config in your `config.yaml`, it will continue to work. Passing backend-specific flags to the CLI is just more elegant :)

File: src/instructlab/model/serve.py [hunk 3]
```diff
@@ -87,9 +78,14 @@ def serve(
     model_family,
     log_file: pathlib.Path | None,
     backend: str | None,
-    vllm_args: typing.Iterable[str],
 ) -> None:
-    """Start a local server"""
+    """Start a local server
+
+    The vLLM backend accepts additional parameters in the form of extra
+    arguments after "--" separator:
```
[the remark this answers is not in the document]
Comment (tiran) [reply src/instructlab/model/serve.py:85]: Good idea! I had to shorten the URL and omit the anchor. click wraps line > 78 chars, which breaks the URL.

**Review by leseb — APPROVED** [review]

**Comment by leseb** [comment]
@n1hility dismissed your review because the PR got enough attention, feel free to comment if you have any concerns with this approach. Thanks!

**Pull request merged**